### PR TITLE
Updated Docs for Hive integration

### PR DESCRIPTION
**Pull request by KJonline**
## Proposed change
This change is to update the hive documentation to support the changes in PR https://github.com/home-assistant/core/pull/45833

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
- [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/45833
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/hive.markdown [hunk 1]
```diff
@@ -23,43 +23,42 @@ This integration uses the unofficial API used in the official Hive website [http
...
+- [Services](#services)
+  - [Service `hive.boost_heating`](#service-hiveboost_heating)
+  - [Service `hive.boost_hot_water`](#service-hiveboost_hot_water)
+- [Platforms](#platforms)
+  - [Binary Sensor](#binary-sensor)
+  - [Climate](#climate)
+  - [Light](#light)
+  - [Sensor](#sensor)
+  - [Switch](#switch)
+  - [Water Heater](#water-heater)
```
Comment (klaasnicolaas) [inline source/_integrations/hive.markdown]: Something goes wrong here, possibly this list can be completely removed due to the TOC on the side of the page.

**Review by klaasnicolaas — CHANGES_REQUESTED** [review]

File: source/_integrations/hive.markdown (hunk 1 above)
Comment (KJonline) [reply source/_integrations/hive.markdown]: Hi @klaasnicolaas  this has been removed in the latest commit

**Review by frenck — DISMISSED** [review]
✅ Approved. Can be merged as soon as the parent PR gets merged.

File: source/_integrations/hive.markdown [hunk 2]
```diff
@@ -23,43 +23,29 @@ This integration uses the unofficial API used in the official Hive website [http
 
 There is currently support for the following platforms within Home Assistant:
```
Comment (klaasnicolaas) [inline source/_integrations/hive.markdown]: Maybe also delete this line 😉

**Review by klaasnicolaas — CHANGES_REQUESTED** [review]

File: source/_integrations/hive.markdown (hunk 2 above)
Comment (KJonline) [reply source/_integrations/hive.markdown]: Removed in the latest Commit.

**Comment by KJonline** [comment]
@klaasnicolaas Changes have been made as suggested.

**Pull request closed without merging**